unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/VideoPlayer/VideoPlayerHandler.cs

```diff
@@ -14,68 +14,120 @@ public class VideoPlayerHandler : IECSComponentHandler<PBVideoPlayer>
+        }
 
         public void OnComponentRemoved(IParcelScene scene, IDCLEntity entity)
         {
+            Helpers.Utils.OnCursorLockChanged -= OnCursorLockChanged;
+            CommonScriptableObjects.sceneNumber.OnChange -= OnSceneChanged;
+            CommonScriptableObjects.rendererState.OnChange -= OnRendererStateChanged;
+
             videoPlayerInternalComponent.RemoveFor(scene, entity);
             videoPlayer?.Dispose();
         }
 
         public void OnComponentModelUpdated(IParcelScene scene, IDCLEntity entity, PBVideoPlayer model)
         {
-            if (lastModel != null && lastModel.Equals(model))
-                return;
-
-            // detect change of the state
+            // Setup video player
             if (lastModel == null || lastModel.Src != model.Src)
             {
                 videoPlayer?.Dispose();
 
                 var id = entity.entityId.ToString();
                 bool isStream = !NO_STREAM_EXTENSIONS.Any(x => model.Src.EndsWith(x));
                 string videoUrl = model.GetVideoUrl(scene.contentProvider, scene.sceneData.requiredPermissions, scene.sceneData.allowedMediaHostnames);
-                videoPlayer = new WebVideoPlayer(id, videoUrl, isStream, DCLVideoTexture.videoPluginWrapperBuilder.Invoke());
 
+                isValidUrl = !string.IsNullOrEmpty(videoUrl);
+                if (!isValidUrl)
+                    return;
+
+                videoPlayer = new WebVideoPlayer(id, videoUrl, isStream, DCLVideoTexture.videoPluginWrapperBuilder.Invoke());
                 videoPlayerInternalComponent.PutFor(scene, entity, new InternalVideoPlayer()
                 {
                     videoPlayer = videoPlayer,
                     assignedMaterials = new List<InternalVideoPlayer.MaterialAssigned>(),
                 });
             }
 
-            // detect change of the state
-            if (lastModel == null || lastModel.IsPlaying() != model.IsPlaying())
-            {
-                if (model.IsPlaying()) { videoPlayer.Play(); }
-                else { videoPlayer.Pause(); }
-            }
-
-            // detect change of the state
+            // Apply model values except 'Playing'
             float lastPosition = lastModel?.GetPosition() ?? 0.0f;
-
             if (Math.Abs(lastPosition - model.GetPosition()) > 0.01f) // 0.01s of tolerance
-            {
                 videoPlayer.SetTime(model.GetPosition());
-            }
-
             videoPlayer.SetVolume(model.GetVolume());
-
             videoPlayer.SetPlaybackRate(model.GetPlaybackRate());
-
             videoPlayer.SetLoop(model.GetLoop());
 
             lastModel = model;
+
+            ConditionsToPlayVideoChanged();
+        }
+
+        private void ConditionsToPlayVideoChanged()
+        {
+            if (lastModel == null) return;
+
+            bool shouldBePlaying = lastModel.IsPlaying() && canVideoBePlayed;
+            if (shouldBePlaying != videoPlayer.playing)
```

Comment:
do we want the "video not playing if in another scene" behavior? the video is already reducing the frequency it updates if user is not in the same scene

Reply:
Removed